integration-tests/validation.py

```diff
@@ -222,5 +224,59 @@ def validate_status_messages(model):
     await wait_for_ready(model)
 
 
+async def validate_api_extra_args(model):
+    app = model.applications['kubernetes-master']
+    original_config = await app.get_config()
+
+    async def get_apiserver_args():
+        results = []
+        for unit in app.units:
+            action = await unit.run('pgrep -a kube-apiserver')
+            assert action.status == 'completed'
+            raw_output = action.data['results']['Stdout']
+            args = raw_output.partition(' ')[2].partition(' ')[2]
+            results.append(args)
+        return results
+
+    original_args = await get_apiserver_args()
+
+    extra_args = ' '.join([
+        'min-request-timeout=314',  # int arg, overrides a charm default
+        'watch-cache=true',         # bool arg, explicit
+        'enable-swagger-ui'         # bool arg, implied true
+    ])
+    await app.set_config({'api-extra-args': extra_args})
+
+    expected_args = [
+        '--min-request-timeout=314',
+        '--watch-cache=true',
+        '--enable-swagger-ui=true'
+    ]
+
+    deadline = time.time() + 60
+    while time.time() < deadline:
+        all_args = await get_apiserver_args()
+        for args in all_args:
+            for expected_arg in expected_args:
+                if expected_arg not in args:
+                    await asyncio.sleep(1)
+                    continue
+        break
+    else:
+        raise TimeoutError('api-extra-args config did not propagate')
+
```

Comment:
Nitpick, but this block could be simplified (at least syntactically) to:

```python
while time.time() < deadline:
    all_args = await get_apiserver_args()
    if set(expected_args) <= set(all_args):
        break
    await asyncio.sleep(1)
else:
    raise TimeoutError('api-extra-args config did not propagate')
```

Reply:
`all_args` is a list of full argument strings for each unit, something like:

```
[
    '--logtostderr=true --etcd-ca-file=/root/cdk/certs/ca.crt --etcd-cert-file=...',
    '--logtostderr=true --etcd-ca-file=/root/cdk/certs/ca.crt --etcd-cert-file=...'
]
```

To do what you're suggesting, we need to `.split()` the arg strings, and validate against all of them:

```
if all(set(expected_args) <= set(args.split()) for args in all_args):
    break
```

Guess I'll do that.

Also, `all_args` is confusingly named. I'mma rename it to `args_for_all_units` or something.